### PR TITLE
Added (de)activate scripts; bump to 0.1.5

### PR DESCRIPTION
**Pull request by dipterix**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

R package `rpymat` creates isolated conda environment for R packages. In the vanilla version, the package downloads conda and create isolated environments. However, if installed within conda, `rpymat` could use the already-installed conda without downloading/installing another copy of conda. The latest version `0.1.5` allows users to specify custom conda path via system environment variables `R_RPYMAT_CONDA_EXE` and `R_RPYMAT_CONDA_PREFIX`. This conda recipe automatically points these variables to the current conda environment.

### CODE REVIEW

**Comment by dipterix** [comment]
@conda-forge-admin, please rerender

**Comment by dipterix** [comment]
@conda-forge/help-r Could you please take one minute to help me check if this PR has any issues from the conda-forge side? I tried to follow instructions to add (de)activate scripts & call conda-smithy to re-render, but might still make mistakes/not follow the conventions. 

Thanks in advance :)

File: recipe/activate.sh
Comment (mfansler) [inline recipe/activate.sh:1]: It would be better to backup and restore (during deactivate) any environment variables, rather than set and unset.

**Review by mfansler — COMMENTED** [review]
I generally do not like anything that requires co-installation of R and Python in the same environment. It is almost always a better end-user experience to partition them modularly. I realize you are the upstream maintainer, but is this strictly necessary?

Other than the `R_RPYMAT_CONDA_EXE` variable, I don't see why the other ones should be auto-set to the current environment. Personally, I want the freedom to have modularly separated R and Python environments, and I can use tools like `reticulate` to connect to Python when I need to without having to install `python` in my R environment.

I'm not saying this has to change - just voicing my opinion that I have reservations about this from a software engineering perspective.

File: recipe/activate.sh
Comment (dipterix) [reply recipe/activate.sh:1]: Gotcha, I'll remove Python from the requirement. As for "backup and restore (during deactivate) any environment variables", is there any existing examples that I can follow, or it's fine as long as I back up and restore the env vars?

Thanks for the comments!

Comment (mfansler) [reply recipe/activate.sh:1]: Here is a clean example: https://github.com/conda-forge/octave-feedstock/tree/main/recipe/scripts

If you search, you can also find similar ones with `.bat` examples.

**Comment by dipterix** [comment]
@conda-forge-admin, please rerender

**Comment by mfansler** [comment]
I believe that since this is a `noarch` build, the `bld.bat` is never run. Instead you may want to also have the `.bat` files copied into place by the .sh script.

**Comment by dipterix** [comment]
> I believe that since this is a `noarch` build, the `bld.bat` is never run. Instead you may want to also have the `.bat` files copied into place by the .sh script.

Would you mind showing me an example/tutorial? I don't know what exactly I should do... (sorry, still learning). Do you mean I should have `build.sh` copy `.bat` files as well? Do I need to check platform or can I copy regardless of the OS?

Also I removed python from deps, used backup+restore for env vars.

**Comment by dipterix** [comment]
@conda-forge-admin, please rerender

**Comment by dipterix** [comment]
@mfansler I added environment variables in `meta.yaml` for windows builders. `build.sh` detects this env var and copies `.bat` files accordingly. Is that legit?

**Comment by mfansler** [comment]
> @mfansler I added environment variables in `meta.yaml` for windows builders. `build.sh` detects this env var and copies `.bat` files accordingly. Is that legit?

No, because the build script only runs on Linux host. Instead, just copy all of them without conditioning on OS, and let the target OS determine which is appropriate to run.

**Comment by mfansler** [comment]
Also, the scripts still set the Conda prefix and Python to point at current environment. I was recommending not setting them at all. I.e., only auto-set the Conda exe. But again, that's a design choice.

**Comment by dipterix** [comment]
> Also, the scripts still set the Conda prefix and Python to point at current environment. I was recommending not setting them at all. I.e., only auto-set the Conda exe. But again, that's a design choice.

The vanilla `rpymat` does not use user-installed conda: it installs conda to an independent place (specified by R `tools::R_user_dir`) and manages a total independent env to maximize reproducibility. This design makes sure that any changes to outside installations of conda does not affect the code. 

The prefix is only used when the R package is installed via `r-rpymat`, which means if the environment is removed, changes set by rpymat should be also removed.

**Pull request merged**